lib/db/lowlevel.go

```diff
@@ -34,6 +37,8 @@ type Lowlevel struct {
 	location  string
 	folderIdx *smallIndex
 	deviceIdx *smallIndex
+	closeMut  *sync.RWMutex
```

Comment:
this could as well be a boolean given that we only look at it from within the mutex?

Reply:
Presumably having a channel here would be easier? Single member variable?

Reply:
"Unprotected" channel is still racy:  

1. `NewIterator` steps past the closed check.  
2. `Close` closes the channel.  

Result: `DB.Close` must happen after `DB.NewIterator`, but may happen before.